// Copyright (c) 2017, 2018, 2019 Chris Cummins.
//
// DeepSmith is free software: you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation, either version 3 of the License, or
// (at your option) any later version.
//
// DeepSmith is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.
//
// You should have received a copy of the GNU General Public License
// along with DeepSmith.  If not, see <https://www.gnu.org/licenses/>.

syntax = "proto2";

package deepsmith;

option go_package = "deepsmithpb";
option java_multiple_files = true;
option java_outer_classname = "ServiceProto";
option java_package = "com.deepsmith";

message ServiceStatus {
  enum ReturnCode {
    SUCCESS = 0;
    ERROR = 1;
    NOT_IMPLEMENTED_ERROR = 2;
    INVALID_REQUEST_PARAMETERS = 3;
  }
  // The name of the server sending the response.
  optional string client = 1 [default = "<anon>"];
  // The return code of the response.
  optional ReturnCode returncode = 2 [default = NOT_IMPLEMENTED_ERROR];
  // If status != SUCCESS, a message describing the error.
  optional string error_message = 3;
}

message ServiceConfig {
  optional string hostname = 1;
  optional int32 port = 2;
}
